source/codegen/templates/service.cpp.mako

```diff
@@ -41,9 +41,13 @@ namespace ${config["namespace_component"]}_grpc {
 %   for custom_type in custom_types:
   void ${service_class_prefix}Service::Copy(const ${custom_type["name"]}& input, ${namespace_prefix}${custom_type["grpc_name"]}* output) 
   {
-%     for field in custom_type["fields"]: 
-    output->set_${field["grpc_name"]}(input.${field["name"]});
-%     endfor
+%for field in custom_type["fields"]: 
+    %if field["type"] == 'NIComplexNumber':
```

Comment:
await: We shouldn't refer to any driver-specific types from this service. Instead of checking for `NIComplexNumber`, you should check to see if the type is found in the `custom_types`. The only exception I can think of would be the session type in session.proto since that's shared by all drivers. But we haven't seen the session in a custom struct yet.

Reply:
Yes, fixed it.